rustio/src/client.rs

```diff
@@ -35,14 +36,15 @@ pub enum ClientUpdate {
     Clear,
 }
 
+#[derive(Clone)]
 pub struct Client {
-    current_search_id: Rc<RefCell<u64>>,
+    current_search_id: Arc<Mutex<u64>>,
```

Comment:
Why the change from Rc/RefCell to Arc/Mutex? I'm missing something?

Reply:
Needed to change it to be able to share the Client structure to the new thread, without those changes I get this
``
thread::spawn( move || {
   |         ^^^^^^^^^^^^^ `std::rc::Rc<std::cell::RefCell<u64>>` cannot be sent between threads safely
``
Thought you wouldn't agree cause each time I use clone multiple time I feel that maybe there's a better way plus didn't really want to change rustio crate at all but this is the solution I found for now.
Thanks for the feedback !

Reply:
Oh, I see. 

I have no problem, if you edit the rustio crate. Both crates are still in development. Everything is fine, as long they are independently to each other.

Currently I'm on holiday, as soon as I'm home again, I'll merge your branch. Thanks for your work!

Reply:
Thank you too I'm learning from your work plus Gradio is a very awesome app !